# Import Software Packages

In [1]:
import sys                                             # Read system parameters.
import os                                              # Interact with the operating system
import numpy as np                                     # Work with multi-dimensional arrays and matrices.
import pandas as pd                                    # Manipulate and analyze data.
import matplotlib as mpl                               # Create 2D charts.
import matplotlib.pyplot as plt                        # Create figures
import matplotlib.cm as cm                             # Create Color maps
import seaborn as sb                                   # Perform data visualization.
import sklearn                                         # Perform data mining and analysis.
from sklearn import datasets

# Summarize software libraries used.
print('Libraries used in this project:')
print('- Python {}'.format(sys.version))
print('- NumPy {}'.format(np.__version__))
print('- pandas {}'.format(pd.__version__))
print('- Matplotlib {}'.format(mpl.__version__))
print('- Seaborn {}'.format(sb.__version__))
print('- scikit-learn {}\n'.format(sklearn.__version__))

Libraries used in this project:
- Python 3.8.18 | packaged by conda-forge | (default, Dec 23 2023, 17:23:49) 
[Clang 15.0.7 ]
- NumPy 1.24.3
- pandas 2.0.3
- Matplotlib 3.7.2
- Seaborn 0.9.0
- scikit-learn 1.3.2



# Load Dataset

In [2]:
PROJECT_ROOT_DIR = "."
DATA_PATH = os.path.join(PROJECT_ROOT_DIR, "data/")
print("Data files in this project:", os.listdir(DATA_PATH))

# Read the raw dataset
raw_housing_data_file = os.path.join(DATA_PATH, "BostonHousing.csv")
data_raw = pd.read_csv(raw_housing_data_file)
print(
    "Loaded {} records from {}.\n".format(len(data_raw), raw_housing_data_file)
)

Data files in this project: ['BostonHousing.csv', '.ipynb_checkpoints']
Loaded 506 records from ./data/BostonHousing.csv.



# Explore the Dataset

#### View features and data types

**View Columns**

- **CRIM** is the per-capita crime rate of the area.
- **CHAS** refers to the "Charles River dummy variable"—if 1, the land bounds the river; if 0, it
does not.
- **NOX** is the level of nitrogen oxides (NO2) in the area. High levels of NO2 can cause health
issues.
- **RM** is the average number of rooms per house.
- **AGE** is the proportion of occupied units built before 1940.
- **DIS** is the weighted mean distance to several employment centers around Boston.
- **TAX** is the property tax rate per $10,000.
- **PTRATIO** is the ratio of pupils (students) to teachers in the school district.
- **Target** also refers to MEDV, the median value of the house in thousands of dollars

In [15]:
data_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 14 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CRIM     506 non-null    float64
 1   ZN       506 non-null    float64
 2   INDUS    506 non-null    float64
 3   CHAS     506 non-null    int64  
 4   NOX      506 non-null    float64
 5   RM       506 non-null    float64
 6   AGE      506 non-null    float64
 7   DIS      506 non-null    float64
 8   RAD      506 non-null    int64  
 9   TAX      506 non-null    int64  
 10  PTRATIO  506 non-null    float64
 11  B        506 non-null    float64
 12  LSTAT    506 non-null    float64
 13  Target   506 non-null    float64
dtypes: float64(11), int64(3)
memory usage: 55.5 KB


- The training set includes **506 rows** and **14 columns**.
- 11 columns contain float values and **3** columns contain integer values.
- There is **no missing data** as all rows have values for every column.